### PR TITLE
[DIR-1444] [DIR-724] [DIR-1506] migrate to new instances api

### PR DESCRIPTION
**Pull request by stefan-kracht**
## Description

Migrating to the v2 instance API

- The instance list endpoint does not support streaming anymore, I added a reload button to the top of the list
- instances can now also be filtered by `status === crashed` (the possible statuses were hard-coded in the filter component, I added a direct binding to the schema so that every new status will automatically be added to the filter)
- the streaming on the details page will now stop when the instance is not running anymore (this saves some browser resources, the streaming endpoint pushes updates very frequently, even when nothing changes)
- found a way to fix [DIR-724](https://linear.app/direktiv/issue/DIR-724/fix-typing-for-filterval-to-a-dynamic-value-like-instanceinstance-id) 
- I updated the API commands button on the workflow page to use the new API (this closes [DIR-1506](https://linear.app/direktiv/issue/DIR-1506/update-api-button-to-use-v2-api))
- I removed the `api/tree` folder after we migrated all APIs (the `FileNameSchema` is the only thing we still use, I moved that into `src/api/files/schema.ts`)
- The new API design fixes a major bug in the diagram on the instance page. The diagram was always generated based on the current file content of the workflow that triggered the instance. This workflow might have changed since the workflow ran, and the diagram could show something completely wrong. If the file was deleted or renamed after the instance was generated, the diagram would not have been displayed at. With the new API, the content of the workflow that created the instance will be part of the response and can be used as a reliable source for the diagram.

## Follow-up tickets

To let us move a bit quicker in the API migration, I added the following task as a separate ticket as I think it is a good iteration and not a mandatory 
- [handle big output on the instance page](https://linear.app/direktiv/issue/DIR-1522/handle-big-instance-output-on-the-instance-page)

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

### CODE REVIEW

**Review by sebxian — COMMENTED** [review]
This looks good to me!

Some comments in the code regarding naming.

I also have one question -- this is probably outside of the scope of this PR, but for clarification: Was there a discussion with the backend guys regarding the format of the filter params? I think they could be improved, and that might also lead to a simplification of currently hard coded stuff in the ui. But that would probably be a separate issue.

File: ui/src/pages/namespace/Monitoring/Instances/index.tsx [hunk 1]
```diff
@@ -47,6 +47,9 @@ export const Instances = () => {
 
   const { t } = useTranslation();
 
+  const sucessfullInstances = dataSucessfullInstances?.data ?? [];
```
[the remark this answers is not in the document]
Comment (stefan-kracht) [reply ui/src/pages/namespace/Monitoring/Instances/index.tsx]: fixed in 3f17935edae4bb06b34547c1808a40d9e9c519c7 (plus a couple of others)

File: ui/src/pages/namespace/Instances/Detail/Header/index.tsx [hunk 2]
```diff
@@ -20,28 +20,28 @@ import useUpdatedAt from "~/hooks/useUpdatedAt";
...
+  const { data: instance } = useInstanceDetails({ instanceId });
+  const { mutate: cancelInstance, isPending } = useCancelInstance();
 
   const { t } = useTranslation();
 
-  const [invoker] = (data?.instance?.invoker ?? "").split(":");
-  const updatedAt = useUpdatedAt(data?.instance.updatedAt);
-  const createdAt = useUpdatedAt(data?.instance.createdAt);
+  const [invoker] = (instance?.invoker ?? "").split(":");
+  const endetAt = useUpdatedAt(instance?.endedAt);
```
[the remark this answers is not in the document]
Comment (stefan-kracht) [reply ui/src/pages/namespace/Instances/Detail/Header/index.tsx]: fixed in eb5896298bbe8d9c72a2ba19866519786d6bdc96

File: ui/src/api/instances/index.ts [hunk 3]
```diff
@@ -20,13 +25,13 @@ export const instanceKeys = {
         filters,
       },
     ] as const,
-  instanceDetail: (
```
[the remark this answers is not in the document]
Comment (stefan-kracht) [reply ui/src/api/instances/index.ts:23]: I don't think we had a convention for cache keys yet. The main reason for my change was to have if consistent with "variables-details" (the only other key that has a list/details pattern)

I agree to your suggestion and changed it accordingly (96b7608627e652ab143b010979c0d5e8cd02b851). I also changed the scope name for the variable details as well (8e9368846f300c85ec7f0d486c7756e3e1bf8514).

File: ui/src/api/instances/query/get.ts [hunk 4]
```diff
@@ -1,120 +1,50 @@
...
-      queryField = "CREATED";
-      queryValue = date.toISOString();
-    } else {
-      const value = filters[field]?.value;
-      if (!value) {
-        throw new Error("filter value is not defined");
-      }
-      queryField = field;
-      queryValue = value;
+export const getInstanceList = apiFactory({
```
[the remark this answers is not in the document]
Comment (stefan-kracht) [reply ui/src/api/instances/query/get.ts]: Done (0b392a3c64081cc3da83bdc248644610f1524cc2)

**Comment by stefan-kracht** [comment]
> This looks good to me!
> 
> Some comments in the code regarding naming.
> 
> I also have one question -- this is probably outside of the scope of this PR, but for clarification: Was there a discussion with the backend guys regarding the format of the filter params? I think they could be improved, and that might also lead to a simplification of currently hard coded stuff in the ui. But that would probably be a separate issue.

No, there was no discussion of changing the query parameters. Maybe we make this is a future ticket and align the params with the event filter ones.

**Review by sebxian — APPROVED** [review]
Thanks for the updates. As discussed, I pushed some more changes to get all the spelling consistent. Good idea about a follow-up for event & instances filters, I'll keep this in mind while working on the events!

**Pull request merged**